src/routes/commercialRegistrations/find.ts

```diff
@@ -0,0 +1,77 @@
+import type { Request, Response } from "express";
+import { businesses } from "../../fixtures/businesses";
+import { ModelNotFoundError } from "./types/modelNotFoundError";
+import { Business } from "./types/business";
+import { findQuery } from "./types/findQuery";
+import { addresses } from "../../fixtures/addresses";
+import { businessNames } from "../../fixtures/businessNames";
+import { germanFirstNames } from "../../fixtures/firstNames";
+import { germanLastNames } from "../../fixtures/lastNames";
+
+export type FindRequest = Partial<Request<{}, {}, {}, findQuery>>;
+export type FindResponse = Partial<Response<Business | ModelNotFoundError>>;
+
+export const modelNotFoundError: ModelNotFoundError = {
+  title: "Model Not Found",
+  status: "404",
+  code: "model_not_found",
+  detail: "someString",
+  id: "someString",
+};
+
+/**
+ * @see {@link https://docs.solarisgroup.com/api-reference/onboarding/businesses/#tag/Business-Registrations/paths/~1v1~1commercial_registrations~1find/get}
+ */
+export const find = (req: FindRequest, res: FindResponse) => {
+  const {
+    registration_number,
+    registration_issuer,
+    country = "DE",
+  } = req.query;
+  const foundBusiness = businesses.find(
+    (business) =>
+      String(business.registration_number).replace(" ", "") ===
+        registration_number &&
+      business.registration_issuer === registration_issuer &&
+      business.address.country === country
+  );
+
+  const shouldGenerateBusiness =
```

Comment:
this request will be always called with full 'registration_number' and 'registration_issuer'
first we call 'search' and it returns an array of suggestions, then with 'registration_number' and 'registration_issuer' we call 'find'

Reply:
This is to make it possible to make a query that results in 404. If you have another idea in mind we can also do that.